### ISSUE TITLE
Can't open

### ISSUE THREAD

**Issue by santaclose** [jeremyong/flop#4]
Downloaded the binary to give it a try but it doesn't open. 
https://user-images.githubusercontent.com/6496455/166145921-a34726c4-708e-4e11-bc93-3a534553292c.mp4

The window goes completely white, the program logs `Using device: NVIDIA GeForce GTX 1070`, and then it exits.

![image](https://user-images.githubusercontent.com/6496455/166146087-74abe6f9-5ce2-4a84-868d-328e2e3d14af.png)
I also tried to pass a test image, but the same happens.

**Comment by jeremyong** [comment]
I'm going to assume the issue is your graphics driver doesn't support some of the feature bits set by the vulkan device on initialization (for example, dynamic render passes). Can you paste the output of running `vulkaninfo.exe`?

**Comment by santaclose** [comment]
[vulkaninfo_output.txt](https://github.com/jeremyong/flop/files/8644816/vulkaninfo_output.txt)

**Comment by jeremyong** [comment]
@santaclose Sorry for the delay again, I'm horrible with github notifications. From what I can tell, the only missing feature bit for your card/driver is `descriptorBindingUniformBufferUpdateAfterBind` which I am setting. However, the good news is, I don't actually need this feature bit. Do you think you'd be able to try compiling and running the latest commit on the main branch?

**Comment by santaclose** [comment]
I tried to follow the build instructions but something went wrong. Also, I don't know if I have the Vulkan SDK installed. Can you upload a binary somewhere?
```
A:\flop (main -> origin)
λ cmake -B build -S .
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.19041.0 to target Windows 10.0.19043.
-- The C compiler identification is MSVC 19.29.30133.0
-- The CXX compiler identification is MSVC 19.29.30133.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Microsoft (R) Build Engine version 16.11.0+0538acc04 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  Checking Build System
  Creating directories for 'dxcompiler-populate'
  Building Custom Rule A:/flop/build/_deps/dxcompiler-subbuild/CMakeLists.txt
  Performing download step (download, verify and extract) for 'dxcompiler-populate'
  -- Downloading...
     dst='A:/flop/build/_deps/dxcompiler-subbuild/dxcompiler-populate-prefix/src/dxc_2021_12_08.zip'
     timeout='none'
     inactivity timeout='none'
  -- Using src='https://github.com/microsoft/DirectXShaderCompiler/releases/download/v1.6.2112/dxc_2021_12_08.zip'
  -- [download 0% complete]
  -- [download 1% complete]
  -- [download 2% complete]
  -- [download 3% complete]
  -- [download 4% complete]
  -- [download 5% complete]
  -- [download 6% complete]
  -- [download 7% complete]
  -- [download 8% complete]
  -- [download 9% complete]
  -- [download 10% complete]
  -- [download 11% complete]
  -- [download 12% complete]
  -- [download 13% complete]
  -- [download 14% complete]
  -- [download 15% complete]
  -- [download 16% complete]
  -- [download 17% complete]
  -- [download 18% complete]
  -- [download 19% complete]
  -- [download 20% complete]
  -- [download 21% complete]
  -- [download 22% complete]
  -- [download 23% complete]
  -- [download 24% complete]
  -- [download 25% complete]
  -- [download 26% complete]
  -- [download 27% complete]
  -- [download 28% complete]
  -- [download 29% complete]
  -- [download 30% complete]
  -- [download 31% complete]
  -- [download 32% complete]
  -- [download 33% complete]
  -- [download 34% complete]
  -- [download 35% complete]
  -- [download 36% complete]
  -- [download 37% complete]
  -- [download 38% complete]
  -- [download 39% complete]
  -- [download 40% complete]
  -- [download 41% complete]
  -- [download 42% complete]
  -- [download 43% complete]
  -- [download 44% complete]
  -- [download 45% complete]
  -- [download 46% complete]
  -- [download 47% complete]
  -- [download 48% complete]
  -- [download 49% complete]
  -- [download 50% complete]
  -- [download 51% complete]
  -- [download 52% complete]
  -- [download 53% complete]
  -- [download 54% complete]
  -- [download 55% complete]
  -- [download 56% complete]
  -- [download 57% complete]
  -- [download 58% complete]
  -- [download 59% complete]
  -- [download 60% complete]
  -- [download 61% complete]
  -- [download 62% complete]
  -- [download 63% complete]
  -- [download 64% complete]
  -- [download 65% complete]
  -- [download 66% complete]
  -- [download 67% complete]
  -- [download 68% complete]
  -- [download 69% complete]
  -- [download 70% complete]
  -- [download 71% complete]
  -- [download 72% complete]
  -- [download 73% complete]
  -- [download 74% complete]
  -- [download 75% complete]
  -- [download 76% complete]
  -- [download 77% complete]
  -- [download 78% complete]
  -- [download 79% complete]
  -- [download 80% complete]
  -- [download 81% complete]
  -- [download 82% complete]
  -- [download 83% complete]
  -- [download 84% complete]
  -- [download 85% complete]
  -- [download 86% complete]
  -- [download 87% complete]
  -- [download 88% complete]
  -- [download 89% complete]
  -- [download 90% complete]
  -- [download 91% complete]
  -- [download 92% complete]
  -- [download 93% complete]
  -- [download 94% complete]
  -- [download 95% complete]
  -- [download 96% complete]
  -- [download 97% complete]
  -- [download 98% complete]
  -- [download 99% complete]
  -- [download 100% complete]
  -- Downloading... done
  -- extracting...
       src='A:/flop/build/_deps/dxcompiler-subbuild/dxcompiler-populate-prefix/src/dxc_2021_12_08.zip'
       dst='A:/flop/build/_deps/dxcompiler-src'
  -- extracting... [tar xfz]
CUSTOMBUILD : CMake error : Problem with archive_write_finish_entry(): Can't restore time [A:\flop\build\_deps\dxcompiler-subbuild\dxcompiler-populate.vcxproj]
  -- extracting... [error clean up]
CUSTOMBUILD : CMake error : Problem extracting tar: A:/flop/build/_deps/dxcompiler-subbuild/dxcompiler-populate-prefix/src/dxc_2021_12_08.zip [A:\flop\build\_deps\dxcompiler-subbuild\dxcompiler-populate.vcxproj]
  CMake Error at dxcompiler-subbuild/dxcompiler-populate-prefix/src/dxcompiler-populate-stamp/extract-dxcompiler-populate.cmake:33 (message): CUSTOMBUILD : error : extract of [A:\flop\build\_deps\dxcompiler-subbuild\dxcompiler-populate.vcxproj]
    'A:/flop/build/_deps/dxcompiler-subbuild/dxcompiler-populate-prefix/src/dxc_2021_12_08.zip'
    failed


C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\MSBuild\Microsoft\VC\v160\Microsoft.CppCommon.targets(241,5): error MSB8066: Custom build for 'A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-mkdir.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-download.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-update.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-patch.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-configure.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-build.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-install.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dd7c1704ab0a64712aac82c92b8f07c1\dxcompiler-populate-test.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\6bfc05bde7332505db09602ec912b388\dxcompiler-populate-complete.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeFiles\dc049993d9bf55cc82306bcd81bb118c\dxcompiler-populate.rule;A:\flop\build\_deps\dxcompiler-subbuild\CMakeLists.txt' exited with code 1. [A:\flop\build\_deps\dxcompiler-subbuild\dxcompiler-populate.vcxproj]

CMake Error at C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1017 (message):
  Build step for dxcompiler failed: 1
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1146:EVAL:2 (__FetchContent_directPopulate)
  C:/Program Files/CMake/share/cmake-3.21/Modules/FetchContent.cmake:1146 (cmake_language)
  shaders/CMakeLists.txt:9 (FetchContent_Populate)


-- Configuring incomplete, errors occurred!
See also "A:/flop/build/CMakeFiles/CMakeOutput.log".

A:\flop (main -> origin)
λ cmake --build build
Microsoft (R) Build Engine version 16.11.0+0538acc04 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

MSBUILD : error MSB1009: Project file does not exist.
Switch: ALL_BUILD.vcxproj

A:\flop (main -> origin)
λ cmake --version
cmake version 3.21.2

CMake suite maintained and supported by Kitware (kitware.com/cmake).

A:\flop (main -> origin)
λ
```

**Comment by God-damnit-all** [comment]
> @santaclose Sorry for the delay again, I'm horrible with github notifications. From what I can tell, the only missing feature bit for your card/driver is `descriptorBindingUniformBufferUpdateAfterBind` which I am setting. However, the good news is, I don't actually need this feature bit. Do you think you'd be able to try compiling and running the latest commit on the main branch?

I'm having the same problem, and I too am having difficulties compiling.

**Comment by jeremyong** [comment]
Sorry for the late response @ImportTaste and @santaclose . I've added binaries for flop in a new release [here](https://github.com/jeremyong/flop/releases/tag/v1.61803). Please let me know if this resolves things for you

**Comment by God-damnit-all** [comment]
> Sorry for the late response @ImportTaste and @santaclose . I've added binaries for flop in a new release [here](https://github.com/jeremyong/flop/releases/tag/v1.61803). Please let me know if this resolves things for you

Still crashes immediately, unfortunately. Identical output to the screenshots in the OP.

**Comment by santaclose** [comment]
it works for me now

**Comment by God-damnit-all** [comment]
Well, I don't know why. Just to make sure Vulkan is working correctly, I ran an image through [waifu2x-ncnn-vulkan](https://github.com/nihui/waifu2x-ncnn-vulkan) and it worked fine.

Still can't get flop to launch though. Doesn't process images in headless mode, either.

**Comment by God-damnit-all** [comment]
I see there's an Application error in Event Viewer involving the crash:
```
Faulting application name: flop.exe, version: 0.0.0.0, time stamp: 0x62a57263
Faulting module name: vulkan-1.dll, version: 1.2.170.0, time stamp: 0x60380f84
Exception code: 0xc0000005
Fault offset: 0x00000000000474da
Faulting process id: 0x16d4
Faulting application start time: 0x01d881c99578ed0a
Faulting application path: D:\Downloads\FLOP\flop.exe
Faulting module path: C:\WINDOWS\SYSTEM32\vulkan-1.dll
Report Id: 1c0abcbd-b9eb-44d2-83de-1d6419bbc3a1
Faulting package full name: 
Faulting package-relative application ID: 
```

**Comment by jeremyong** [comment]
@ImportTaste can you copy-paste the output of the vulkaninfo command here as well? Chances are there is a similar driver or device incompatibility

**Comment by God-damnit-all** [comment]
Made a backup of vulkan-1.dll and started experimenting with other vulkan-1.dll files on my computer, this one comes from the Edge browser:
```
Faulting application name: flop.exe, version: 0.0.0.0, time stamp: 0x62a57263
Faulting module name: vulkan-1.dll, version: 1.0.1111.2222, time stamp: 0x62a905e0
Exception code: 0xc0000409
Fault offset: 0x000000000006aa8d
Faulting process id: 0x1f70
Faulting application start time: 0x01d881ca76b6df4d
Faulting application path: L:\Downloads\FLOP\flop.exe
Faulting module path: C:\WINDOWS\SYSTEM32\vulkan-1.dll
Report Id: a7f2be7a-c5b7-4598-996c-f5fbe3d696f7
Faulting package full name: 
Faulting package-relative application ID: 
```

This one comes from Discord:
```
Faulting application name: flop.exe, version: 0.0.0.0, time stamp: 0x62a57263
Faulting module name: vulkan-1.dll, version: 1.0.1111.2222, time stamp: 0x62a905e0
Exception code: 0xc0000409
Fault offset: 0x000000000006aa8d
Faulting process id: 0x1f70
Faulting application start time: 0x01d881ca76b6df4d
Faulting application path: L:\Downloads\FLOP\flop.exe
Faulting module path: C:\WINDOWS\SYSTEM32\vulkan-1.dll
Report Id: a7f2be7a-c5b7-4598-996c-f5fbe3d696f7
Faulting package full name: 
Faulting package-relative application ID: 
```

This one comes from VSCodium:
```
Faulting application name: flop.exe, version: 0.0.0.0, time stamp: 0x62a57263
Faulting module name: vulkan-1.dll, version: 1.0.1111.2222, time stamp: 0x62979612
Exception code: 0xc0000409
Fault offset: 0x0000000000064971
Faulting process id: 0x42dc
Faulting application start time: 0x01d881cab799c22a
Faulting application path: L:\Downloads\FLOP\flop.exe
Faulting module path: C:\WINDOWS\SYSTEM32\vulkan-1.dll
Report Id: f0b72418-6f4f-46a0-a2d2-ae162cc17968
Faulting package full name: 
Faulting package-relative application ID: 
```


> @ImportTaste can you copy-paste the output of the vulkaninfo command here as well? Chances are there is a similar driver or device incompatibility

I restored the proper vulkan-1.dll and then ran `vulkaninfo >vulkaninfo.txt` - this was the stderr output, not sure if it's relevant:
```
WARNING: [Loader Message] Code 0 : ReadDataFilesInRegistry: Registry lookup failed to get layer manifest files.
```

Here is the stdout: [vulkaninfo.txt](https://github.com/jeremyong/flop/files/8922657/vulkaninfo.txt)

**Comment by jeremyong** [comment]
Thanks for the report. Later today I will cut a new separate issue and release a debug version that enables validation to give us a bit more output.

**Comment by jeremyong** [comment]
Closing this issue as resolved, but opened #5 to investigate the other problem
